File: 351001/orsik/publisher/src/main/java/com/sergey/orsik/entity/Creator.java
package com.sergey.orsik.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "tbl_creator")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Creator implements Identifiable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 64, unique = true)
    private String login;

    @Column(length = 128)
    private String password;

    @Column(length = 64)
    private String firstname;

    @Column(length = 64)
    private String lastname;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 16)
    private CreatorRole role;
}

File: 351004/Narivonchik/rest1/rest1/src/core/entities/Creator.cs
namespace rest1.core.entities;

public class Creator(
    string login,
    string password,
    string firstname,
    string lastname) : Entity
{
    public string Login { get; set; } = login;

    public string Password { get; set; } = password;

    public string Firstname { get; set; } = firstname;

    public string Lastname { get; set; } = lastname;
    
    public UserRole Role { get; set; } = UserRole.CUSTOMER;
    
    public IEnumerable<News>? News { get; set; }
}